Issue trying to compile Arch Linux

Hi everyone, I am an Arch user and I'm trying to compile the github version of veracrypt, but I have issues when I do the first "make".

When it gets to the Cipher.cpp file to compile, there's a lot of errors (seems linked to declarations ?) but I can't find if anyone else has the same issue. I tried Update 7 and Update 4, but same result and errors.

I save the error prompt in a file joined (Sorry for the french in the file, I don't know how to get errors in English with make on a French computer).

Can someone else me to find why I can't compile ? I don't know where to start to resolve this issue.

Thanks
[log_compile.txt](https://github.com/veracrypt/VeraCrypt/files/6692396/log_compile.txt)

Same here trying to compile VeraCrypt on Fedora 34 (had no issues on Fedora 31, though):

```
Compiling Cipher.cpp
In file included from Cipher.cpp:17:
/home/asc/VeraCrypt/src/Crypto/Twofish.h:58:62: error: reference to ‘byte’ is ambiguous
   58 | d twofish_encrypt_blocks(TwofishInstance *instance, const byte* in_blk, byte* out_blk, uint32 blockCount);
      |                                                           ^~~~

In file included from /home/asc/VeraCrypt/src/Platform/PlatformBase.h:16,
                 from /home/asc/VeraCrypt/src/Platform/Platform.h:16,
                 from Cipher.cpp:13:
/usr/include/c++/11/cstddef:69:14: note: candidates are: ‘enum class std::byte’
   69 |   enum class byte : unsigned char {};
      |              ^~~~
In file included from /home/asc/VeraCrypt/src/Platform/Memory.h:19,
                 from /home/asc/VeraCrypt/src/Platform/Buffer.h:17,
                 from /home/asc/VeraCrypt/src/Platform/Platform.h:17,
                 from Cipher.cpp:13:
/home/asc/VeraCrypt/src/Common/Tcdefs.h:118:17: note:                 ‘typedef uint8_t byte’
  118 | typedef uint8_t byte;
      |                 ^~~~
```
Problem is the new GNU Toolchain in Fedora release 34 - namely the switch from GCC 10 to 11 (https://fedoraproject.org/wiki/Changes/GNUToolchain). Starting with GCC 11 the default for parameter -std is  `-std=gnu++17` instead of `-std=gnu++14`.

Adding this parameter to `Makefile` in line 203 fixes the issue:

`C_CXX_FLAGS += -fdata-sections -ffunction-sections -fpie -std=gnu++14`

I hope this helps.

--- a/src/Makefile
+++ b/src/Makefile
@@ -174,6 +174,15 @@ ifeq "$(shell uname -s)" "Linux"
 
 	PLATFORM := Linux
 	C_CXX_FLAGS += -DTC_UNIX -DTC_LINUX
+
+	# GNU GCC version 11 and higher compile with -std=gnu++17 by default
+        # which breaks "byte" definitions in Crypto++ library. So set 
+        # -std=gnu++14 instead.
+        GCC11PLUS := $(shell expr `$(CC) -dumpversion | cut -f1 -d.` \>= 11)
+        ifeq "$(GCC11PLUS)" "1"
+                CXXFLAGS += -std=gnu++14
+        endif
+
 	
 	ifeq "$(SIMD_SUPPORTED)" "1"
 		CFLAGS += -msse2

